### PR TITLE
OH2-189:  Add page info to paginated API

### PR DESCRIPTION
**Pull request by ArnaudFonzam**
See OH2-189.

### CODE REVIEW

**Review by mwithi — APPROVED** [review]
LGTM. Thanks a lot!

File: src/main/java/org/isf/examination/rest/ExaminationController.java [hunk 1]
```diff
@@ -236,17 +238,19 @@ public ResponseEntity<PatientExaminationDTO> getLastByPatientId(@PathVariable In
...
-	public ResponseEntity<List<PatientExaminationDTO>> getLastNByPatID(@RequestParam Integer limit, @RequestParam Integer patId) throws OHServiceException {
+	public ResponseEntity<PagedResponseDTO<PatientExaminationDTO>> getLastNByPatID(@RequestParam Integer limit, @RequestParam Integer patId) throws OHServiceException {
+		LOGGER.info("Get examinations limit: {}", limit);
+		PagedResponse<PatientExamination> patientExaminationListPageable = examinationBrowserManager.getLastNByPatIDPageable(patId, limit);
 
-		List<PatientExamination> patientExaminationList = examinationBrowserManager.getLastNByPatID(patId, limit);
-
-		if (patientExaminationList == null || patientExaminationList.isEmpty()) {
+		if (patientExaminationListPageable == null || patientExaminationListPageable.getData().isEmpty()) {
+			LOGGER.info("the list is empty");
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/examination/rest/ExaminationController.java:246]:
```suggestion
			LOGGER.info("The patient list is empty.");
```

File: src/main/java/org/isf/patient/rest/PatientController.java [hunk 2]
```diff
@@ -137,17 +139,19 @@ ResponseEntity<PatientDTO> updatePatient(@PathVariable int code, @RequestBody Pa
...
 		LOGGER.info("Get patients page: {}  size: {}", page, size);
-		List<Patient> patients = patientManager.getPatient(page, size);
-		List<PatientDTO> patientDTOS = patients.stream().map(pat -> {
-			return patientMapper.map2DTO(pat);
-		}).collect(Collectors.toList());
-		if (patientDTOS.isEmpty()) {
-			return ResponseEntity.status(HttpStatus.NO_CONTENT).body(patientDTOS);
+		PagedResponse<Patient> patients = patientManager.getPatientsPageable(page, size);
+		if (patients.getData().isEmpty()) {
+			LOGGER.info("the list is empty");
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/patient/rest/PatientController.java:147]:
```suggestion
			LOGGER.info("The patient list is empty.");
```

**Review by dbmalkovsky — APPROVED** [review]
LGTM (I made two small suggestions)

**Comment by mwithi** [comment]
it seems also that method `whet_get_patients_non_parameters_then_return_list_of_PatientDTO_page_0_default_size_and_OK` is failing in checks

**Review by mwithi — APPROVED** [review]
LGTM. Thanks a lot!

**Pull request merged**